operator/src/main/java/org/keycloak/operator/controllers/KeycloakDistConfigurator.java

```diff
@@ -86,6 +87,9 @@ void configureHostname() {
                 .mapOption("hostname-admin-url", HostnameSpec::getAdminUrl)
                 .mapOption("hostname-strict", HostnameSpec::isStrict)
                 .mapOption("hostname-strict-backchannel", HostnameSpec::isStrictBackchannel);
+        optionMapper(Function.identity())
+                .mapOption("proxy", kc -> (!CRDUtils.isTlsConfigured(kc)
+                        && KeycloakIngressDependentResource.isOpenshiftDefaultIngressEnabled(kc)) ? "edge" : null);
```

Comment:
> We don't have (and for now don't want) a first-class citizen proxy field since we might not need it at all soon after the possible hostname config redesign.

Right, it should be a "second class" or inferred setting - it can be moved to a lower priority that additionalOptions.